src/ws/test-channelresponse.c

```diff
@@ -541,6 +541,15 @@ test_resource_failure (TestResourceCase *tc,
   g_assert (cockpit_pipe_get_pid (tc->pipe, &pid));
   g_assert_cmpint (pid, >, 0);
   int pid_fd = syscall(SYS_pidfd_open, pid, 0);
+  if (pid_fd < 0)
+    {
+      if (errno == ENOSYS)
+	g_test_skip ("no pidfd_open support, skipping");
+      else
+        g_error ("pidfd_open call failed: %m");
+
+      return;
```

Comment:
This is slightly "meh"/confusing, as g_error() aborts. It'd be a bit cleaner to move this into the "then" branch. But it doesn't matter in practice (not sure how strongly @allisonkarlitskaya feels about that). Thanks!

Reply:
Ah, I guess it's only logical when you don't know the glib api